src/main/java/net/dv8tion/jda/core/utils/WidgetUtil.java

```diff
@@ -0,0 +1,685 @@
+            /**
+            * The URL for the user's avatar image
+            * If they do not have an avatar set, this will return the URL of their
+            * default avatar
+            * 
+            * @return
+            *      Never-null String containing the member's effective avatar url.
+            */
+            public String getEffectiveAvatarUrl()
+            {
+                return getAvatarUrl() == null ? getDefaultAvatarUrl() : getAvatarUrl();
+            }
+            
+            /**
+             * Gets the nickname of the member. If they do not have a nickname on
+             * the guild, this will return null;
+             * 
+             * @return possibly-null String containing the nickname of the member
+             */
+            public String getNickname()
+            {
+                return nickname;
+            }
+            
+            /**
+             * Gets the visible name of the member. If they have a nickname set,
+             * this will be their nickname. Otherwise, it will be their username.
+             * 
+             * @return never-null String containing the member's effective (visible) name
+             */
+            public String getEffectiveName()
+            {
+                return nickname == null ? username : nickname;
+            }
+            
+            /**
+             * Gets the online status of the member. The widget does not show
+             * offline members, so this status should never be offline
+             * 
+             * @return the {@link net.dv8tion.jda.core.OnlineStatus OnlineStatus} of the member
+             */
+            public OnlineStatus getOnlineStatus()
+            {
+                return status;
+            }
+            
+            /**
+            * The game that the member is currently playing.
+            * If the user is not currently playing a game, this will return null.
+            *
+            * @return
+            *      Possibly-null {@link net.dv8tion.jda.core.entities.Game Game} containing the game that the member is currently playing.
+            */
+            public Game getGame()
+            {
+                return game;
+            }
+            
+            /**
+             * The current voice state of the member.
+             * If the user is not in voice, this will return null.
+             * 
+             * @return possibly-null VoiceState of the member
+             */
+            public VoiceState getVoiceState()
+            {
+                return state;
+            }
+
+            @Override
+            public String toString()
+            {
+                return "WM:" + getName() + '(' + getId() + ')';
+            }
+            
+        }
+        
+        
+        public class Channel
+        {
+            private final int position;
+            private final String id;
+            private final String name;
+            private final List<Member> members;
```

Comment:
If this is the case then we need to specify that these are VoiceChannels, not Channels imo